llvm/include/llvm/IR/IntrinsicsAMDGPU.td

```diff
@@ -2801,6 +2803,32 @@ def int_amdgcn_wmma_bf16_16x16x16_bf16 : AMDGPUWmmaIntrinsicOPSEL<llvm_anyint_ty
 // GFX12 Intrinsics
 //===----------------------------------------------------------------------===//
 
+def int_amdgcn_ds_bvh_stack_push4_pop1_rtn : IntDSBVHStackRtn;
+
+def int_amdgcn_ds_bvh_stack_push8_pop1_rtn :
+  Intrinsic<
+    [llvm_i32_ty, llvm_i32_ty], // %vdst, %addr
+    [
+      llvm_i32_ty,   // %addr
```

Comment:
Why isn't this a pointer type? Same for the rest

Reply:
This was just based on how `int_amdgcn_ds_bvh_stack_rtn` was implemented (L2721). It also used `llvm_i32_ty`